bigtable/admin.go

```diff
@@ -119,6 +120,76 @@ func (ac *AdminClient) backupPath(cluster, backup string) string {
 	return fmt.Sprintf("projects/%s/instances/%s/clusters/%s/backups/%s", ac.project, ac.instance, cluster, backup)
 }
 
+// EncryptionInfo represents the encryption info of a table.
+type EncryptionInfo struct {
+	// TODO: compare to https://github.com/googleapis/java-bigtable/pull/656/files#diff-10655a12f2a8430533dca76435f6829a1a0bc978ada1295529b59a46afbcd95fR28
+	EncryptionStatus *status.Status // TODO: should this be wrapped as EncryptionType has been?
+	EncryptionType   EncryptionType
+	KMSKeyVersion    string
+}
+
+func newEncryptionInfo(pbInfo *btapb.EncryptionInfo) EncryptionInfo {
+	info := EncryptionInfo{}
+	info.EncryptionStatus = pbInfo.EncryptionStatus
+	// TODO: could also just return this as a string, but wrapped
+	info.EncryptionType = EncryptionType(pbInfo.EncryptionType.Number())
+	info.KMSKeyVersion = pbInfo.KmsKeyVersion
+
+	return info
+}
+
+type EncryptionType int32
+
+const (
+	// Encryption type was not specified, though data at rest remains encrypted.
+	ENCRYPTION_TYPE_UNSPECIFIED EncryptionType = 0
+	// The data backing this resource is encrypted at rest with a key that is
+	// fully managed by Google. No key version or status will be populated.
+	// This is the default state.
+	GOOGLE_DEFAULT_ENCRYPTION EncryptionType = 1
+	// The data backing this resource is encrypted at rest with a key that is
+	// managed by the customer.
+	// The in-use version of the key and its status are populated for
+	// CMEK-protected tables.
+	// CMEK-protected backups are pinned to the key version that was in use at
+	// the time the backup was taken. This key version is populated but its
+	// status is not tracked and is reported as `UNKNOWN`.
+	CUSTOMER_MANAGED_ENCRYPTION EncryptionType = 2
+)
+
+// Gets the current encryption info for the table across all of the clusters.
+// The returned map will be keyed by cluster id and contain a status for all of the keys in use.
+func (ac *AdminClient) EncryptionInfo(ctx context.Context, table string) (map[string][]EncryptionInfo, error) {
+	ctx = mergeOutgoingMetadata(ctx, ac.md)
+
+	res, err := ac.getTable(ctx, table)
+	if err != nil {
+		return nil, err
+	}
+	// TODO: backups also needs this.
+	// TODO: why not just return the clusterstates, as it has a map of encryption info.
+	encryptionInfo := map[string][]EncryptionInfo{}
+	for key, cs := range res.ClusterStates {
+		// TODO: if we didn't wrap EncryptionInfo, this could reduce to
+		// returning map[string][]*btapb.EncryptionInfo directly
+		// encryptionInfo[key] = cs.EncryptionInfo
+		encInfo := cs.GetEncryptionInfo()
+		if encInfo == nil {
+			return nil, nil
```

Comment:
Should this return an error or value?

Reply:
Looking now it probably should be deleted :)